src/librustdoc/config.rs

```diff
@@ -476,6 +487,9 @@ impl Options {
         let generate_search_filter = !matches.opt_present("disable-per-crate-search");
         let persist_doctests = matches.opt_str("persist-doctests").map(PathBuf::from);
         let generate_redirect_pages = matches.opt_present("generate-redirect-pages");
+        let runtool = matches.opt_str("runtool");
+        let runtool_args = matches.opt_strs("runtool-arg");
```

Comment:
I'm a bit confused by your first paragraph, it is my understanding that `opt_strs` returns a vec of all arguments passed to (possibly many) `--runtool-arg`'s. The intention being that arguments get passed one at a time, one per `--runtool-arg`. 

As for the question of integration with bootstrap, I am working on that now, but wanted to get the ball rolling here in rustdoc. My plan is to add an unstable flag to bootstrap that makes it pass `--Zdoctest-xcompile` to cargo test invocations. The current state is that compiletests do get cross-compiled, and it already has a `--runtool` option. My change to compiletest removes that option and instead uses Cargo to create the merged `.cargo/config` for a directory and use the runner specified there, with the idea being that it will ensure consistent testing across rustdoc and compiltest.

Reply:
I see now what you meant in your first paragraph. You're right it would be more robust, I just omitted any error checking there, since worst case scenario one string that is in reality several instructions would just get passed wholesale to the runtool (which I think would not in the end be any different from adding them one at a time). 
Also as far as who to ask, I don't know, but I welcome any suggestions since this seems to be pretty central to how the whole rust compiler building and testing process works.

Reply:
> (which I think would not in the end be any different from adding them one at a time)

Since we would be building up a Command instance with these args, the space-splitting wouldn't happen - the recipient would receive the blob in one "argument", as if it had all been quoted together.

One option would be to look at how many `--runtool-arg` arguments there are and treat it differently based on whether there are one or more than one. If there's only one argument, we could do space-splitting on that, and if there is more than one argument, we can leave them as-is. This somewhat mirrors [Cargo's own parsing of the relevant configuration item](https://github.com/rust-lang/cargo/blob/807429e1b6da4e2ec52488ef2f59e77068c31e1f/src/cargo/util/config.rs#L509-L529), which whitespace-splits it is it's just a string, and leaves them alone if given a list. (This can backfire if someone tried to give a single argument with spaces in it, though... :thinking:)

Reply:
It's not possible to correctly space-split command-line arguments based on some heuristic if you also want to support arguments with spaces in them. Just let the `cargo`/the user take care of this. Most people won't be running `rustdoc` directly this way anyway.